plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/MockThriftHiveMetastoreHandler.java

```diff
@@ -0,0 +1,1699 @@
+import io.trino.hive.thrift.metastore.TxnAbortedException;
+import io.trino.hive.thrift.metastore.TxnOpenException;
+import io.trino.hive.thrift.metastore.Type;
+import io.trino.hive.thrift.metastore.UniqueConstraintsRequest;
+import io.trino.hive.thrift.metastore.UniqueConstraintsResponse;
+import io.trino.hive.thrift.metastore.UnknownDBException;
+import io.trino.hive.thrift.metastore.UnknownPartitionException;
+import io.trino.hive.thrift.metastore.UnknownTableException;
+import io.trino.hive.thrift.metastore.UnlockRequest;
+import io.trino.hive.thrift.metastore.WMAlterPoolRequest;
+import io.trino.hive.thrift.metastore.WMAlterPoolResponse;
+import io.trino.hive.thrift.metastore.WMAlterResourcePlanRequest;
+import io.trino.hive.thrift.metastore.WMAlterResourcePlanResponse;
+import io.trino.hive.thrift.metastore.WMAlterTriggerRequest;
+import io.trino.hive.thrift.metastore.WMAlterTriggerResponse;
+import io.trino.hive.thrift.metastore.WMCreateOrDropTriggerToPoolMappingRequest;
+import io.trino.hive.thrift.metastore.WMCreateOrDropTriggerToPoolMappingResponse;
+import io.trino.hive.thrift.metastore.WMCreateOrUpdateMappingRequest;
+import io.trino.hive.thrift.metastore.WMCreateOrUpdateMappingResponse;
+import io.trino.hive.thrift.metastore.WMCreatePoolRequest;
+import io.trino.hive.thrift.metastore.WMCreatePoolResponse;
+import io.trino.hive.thrift.metastore.WMCreateResourcePlanRequest;
+import io.trino.hive.thrift.metastore.WMCreateResourcePlanResponse;
+import io.trino.hive.thrift.metastore.WMCreateTriggerRequest;
+import io.trino.hive.thrift.metastore.WMCreateTriggerResponse;
+import io.trino.hive.thrift.metastore.WMDropMappingRequest;
+import io.trino.hive.thrift.metastore.WMDropMappingResponse;
+import io.trino.hive.thrift.metastore.WMDropPoolRequest;
+import io.trino.hive.thrift.metastore.WMDropPoolResponse;
+import io.trino.hive.thrift.metastore.WMDropResourcePlanRequest;
+import io.trino.hive.thrift.metastore.WMDropResourcePlanResponse;
+import io.trino.hive.thrift.metastore.WMDropTriggerRequest;
+import io.trino.hive.thrift.metastore.WMDropTriggerResponse;
+import io.trino.hive.thrift.metastore.WMGetActiveResourcePlanRequest;
+import io.trino.hive.thrift.metastore.WMGetActiveResourcePlanResponse;
+import io.trino.hive.thrift.metastore.WMGetAllResourcePlanRequest;
+import io.trino.hive.thrift.metastore.WMGetAllResourcePlanResponse;
+import io.trino.hive.thrift.metastore.WMGetResourcePlanRequest;
+import io.trino.hive.thrift.metastore.WMGetResourcePlanResponse;
+import io.trino.hive.thrift.metastore.WMGetTriggersForResourePlanRequest;
+import io.trino.hive.thrift.metastore.WMGetTriggersForResourePlanResponse;
+import io.trino.hive.thrift.metastore.WMValidateResourcePlanRequest;
+import io.trino.hive.thrift.metastore.WMValidateResourcePlanResponse;
+import org.apache.thrift.TException;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+class MockThriftHiveMetastoreHandler
```

Comment:
You can use a proxy . No need for this extra code.

See for reference
- https://github.com/trinodb/trino/blob/023f8b4cb5fc83615aae8cb30da31ffd61004603/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueTableOperationsInsertFailure.java#L71-L83
- https://github.com/trinodb/trino/blob/023f8b4cb5fc83615aae8cb30da31ffd61004603/plugin/trino-raptor-legacy/src/main/java/io/trino/plugin/raptor/legacy/util/DatabaseUtil.java#L49-L59